nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -98,6 +99,32 @@ public MongoDBDirectory getDirectory() {
         return (MongoDBDirectory) directory;
     }
 
+    @Override
+    public DocumentModel getEntryFromSource(String id, boolean fetchReferences) throws DirectoryException {
+        String idFieldName = directory.getSchemaFieldMap().get(getIdField()).getName().getPrefixedName();
+        DocumentModelList result = doQuery(Collections.singletonMap(idFieldName, id), Collections.emptySet(),
+                Collections.emptyMap(), true, -1, 0, false);
+
+        if (result.isEmpty()) {
+            return null;
+        }
+
+        DocumentModel docModel = result.get(0);
+
+        if (isMultiTenant()) {
+            // check that the entry is from the current tenant, or no tenant
+            // at all
+            String tenantId = getCurrentTenantId();
+            if (StringUtils.isNotBlank(tenantId)) {
+                String entryTenantId = (String) docModel.getProperty(schemaName, TENANT_ID_FIELD);
+                if (StringUtils.isNotBlank(entryTenantId) && !entryTenantId.equals(tenantId)) {
+                    return null;
```

Comment:
By design we don't need to check that as it's a filter of query in `doQuery`. Can we remove it ?